announce.go

```diff
@@ -269,13 +307,53 @@ func (a *Announce) nodeContactor() {
 			break
 		}
 		go txRes.run()
+		time.Sleep(time.Millisecond * 100)
+	}
+}
+
+func (a *Announce) logic(tx *stm.Tx) interface{} {
+	inFlightGetPeers := atomic.LoadInt64(&a.inFlightGetPeers)
+	var mode dhtMode
+	nowMode := tx.Get(a.mode).(dhtMode)
+	mode = nowMode
+	if nowMode == get_peers && inFlightGetPeers >= 3 {
+		a.server.logger().Printf("dht Changed mode to: get_peers_stall")
+		mode = get_peers_stall
+	}
+	if nowMode == get_peers_stall && inFlightGetPeers < 1 {
+		mode = get_peers
+		a.server.logger().Printf("dht Changed mode to: from get_peers_stall to get_peers")
+	}
+
+	l := tx.Get(a.pendingAnnouncePeers).(stmutil.List)
+	if l.Len() > 8 {
```

Comment:
I don't think this is enough. The pending announce peers need to be sorted, and we should terminate when the best pending contact is further away than the worst of the 8 announce peers.

Reply:
You are right! we should rewrite the transaction like a sorted map fo Items like:
type Item Struct {
     InfoHash
     Node
     Status with a point value that goes from GOOD to TOBEDELETED
}

Reply:
I managed to achieve this in the branch I refer to.